### DISCUSSION TITLE
Training updating model in many cycles of training

### DISCUSSION THREAD

**Discussion by johanenforcer** [ultralytics/yolov5#2879] [Q&A]
Right now, I have made the training procedure in many cycles of training, meaning that I am rerunning the training process but with the previously trained model as the next model to train in the many cycles of training. My problem is that I am not quite sure exactly what I need to change in order to make this fully work. What I have done currently is that I load the model variable as the current model like so:

![picture](https://user-images.githubusercontent.com/64362830/115561263-b1298280-a2b5-11eb-86bc-18e4f12d58cb.png)

The most important thing is that the model variable gets updated but am I missing updating other important variables also, like ckpt maybe or is doing what I have been doing fine?

Thanks in advance!

**Comment by glenn-jocher** [comment]
@johanenforcer you can start training from previously trained models with no code changes, i.e.:

```shell
$ python train.py --epochs 300 --weights yolov5s.pt
$ python train.py --epochs 300 --weights runs/train/exp/weights/best.pt
$ python train.py --epochs 300 --weights runs/train/exp2/weights/best.pt
$ python train.py --epochs 300 --weights runs/train/exp3/weights/best.pt
$ python train.py --epochs 300 --weights runs/train/exp4/weights/best.pt
$ etc...
```

Each training run will start from the previous training runs best checkpoint.

**Reply by johanenforcer** [reply]
Thanks for telling me, the reason why I went for the code changes was because at the end of running the cycles I need to create a mAP/% of dataset graph, which would be easier if I can make it all run in one session. But do you think adding code changes could still work?

**Reply by johanenforcer** [reply]
In doing as you have proposed, running evolution wouldnt be a problem either since one could just perpetually run the evolve.txt file right?

**Reply by glenn-jocher** [reply]
@johanenforcer yes of course, you can customize the code all you want, but due to high demand and limited resources we can't help you in case of problems with custom code. Community support is always available though.
